examples/gallery/main.cpp

```diff
@@ -36,40 +36,46 @@ int main()
 
     demo->global<TableViewPageAdapter>().set_row_data(row_data);
 
-    demo->global<TableViewPageAdapter>().on_sort_ascending([row_data,
-                                                            demo = slint::ComponentWeakHandle(
-                                                                    demo)](int index) {
-        auto demo_lock = demo.lock();
-        (*demo_lock)
-                ->global<TableViewPageAdapter>()
-                .set_row_data(std::make_shared<slint::SortModel<
-                                      std::shared_ptr<slint::Model<slint::StandardListViewItem>>>>(
-                        row_data, [index](auto lhs, auto rhs) {
-                            auto c_lhs = lhs->row_data(index);
-                            auto c_rhs = rhs->row_data(index);
-
-                            return c_lhs->text < c_rhs->text;
-                        }));
-    });
-
-    demo->global<TableViewPageAdapter>().on_sort_descending([row_data,
-                                                             demo = slint::ComponentWeakHandle(
-                                                                     demo)](int index) {
-        auto demo_lock = demo.lock();
-        (*demo_lock)->global<TableViewPageAdapter>().set_row_data(row_data);
-
-        (*demo_lock)
-                ->global<TableViewPageAdapter>()
-                .set_row_data(std::make_shared<slint::SortModel<
-                                      std::shared_ptr<slint::Model<slint::StandardListViewItem>>>>(
-                        (*demo_lock)->global<TableViewPageAdapter>().get_row_data(),
-                        [index](auto lhs, auto rhs) {
-                            auto c_lhs = lhs->row_data(index);
-                            auto c_rhs = rhs->row_data(index);
-
-                            return c_rhs->text < c_lhs->text;
-                        }));
-    });
+    demo->global<TableViewPageAdapter>().on_filter_sort_model(
+            [](std::shared_ptr<
+                       slint::Model<std::shared_ptr<slint::Model<slint::StandardListViewItem>>>>
+                       source_model,
+               slint::SharedString filter, int sort_index, bool sort_ascending)
+                    -> std::shared_ptr<slint::Model<
+                            std::shared_ptr<slint::Model<slint::StandardListViewItem>>>> {
+                auto model = source_model;
+
+                if (!filter.empty()) {
+                    auto l_filter =
+                            static_cast<std::string>(std::string_view(filter.to_lowercase()));
```

Comment:
There is no need to allocate an extra std::string. Since SharedString has a conversion operator to string_view, it can be used directly with functions like std::string::find() that take a string_view.

```suggestion
                    auto l_filter = filter.to_lowercase());
```